### PR TITLE
63   implement event element matrix

### LINKED ISSUE #63: Implement Event-Element matrix

### PR DESCRIPTION
**Pull request by Exclaminator**
Will close #63

### CODE REVIEW

**Comment by mpsijm** [comment]
Should there be tests written for this class?

File: src/octopeer-github/main/Options/Options.ts [hunk 1]
```diff
@@ -14,6 +14,11 @@ const Options = new (class Options {
     private username: Boolean;
     private repo: Boolean;
     private file: Boolean;
+    private doNotWatchOnScreenEvents: Boolean;
+    private doNotWatchHoverEvents: Boolean;
+    private doNotWatchCommentElements: Boolean;
+    private doNotWatchKeyboardShortcutEvents: Boolean;
```
Comment (mpsijm) [inline src/octopeer-github/main/Options/Options.ts]: I'm assuming these new options should still be implemented in the Options page? If so, please add a `TODO` marker.

**Comment by Exclaminator** [comment]
@mpsijm do you have any suggestions on how to mock the Options class?

**Comment by mpsijm** [comment]
Since `Options` is an object, you can use jasmine spies to mock it. `spyon(Options, "getOptionName")` should do the trick :)

**Comment by Exclaminator** [comment]
Will add tests for the DoNotWatch class

**Comment by mdingjan** [comment]
@Exclaminator I've already made a setup for DoNotWatch tests. 
Actually this should already have been done in the previous week, when the DoNotWatch class was created. I suggest to let me finalize the DoNotWatchTest in #67 , as soon as this PR is merged. 
If you're going to add tests here as well, we will get merge conflicts and you will spend time on something that has already been covered for a large part.

**Comment by Exclaminator** [comment]
Cool, then consider the pr open again!

Met vriendelijke groet,
Youri Arkesteijn

**Pull request closed without merging**